/* ┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
 * ┃     ┏━┣━┣┓　　┏┓┏┓┳┓┏━━┓┣┣━┓　┓　┓┣┳━┓       ┃
 * ┃     ┏┏┏╯━┓┳┳━┛┏╯┃┃┃　　┃┣┣━┓┃┃　┃┃┃　　       ┃
 * ┃     ┃┃┏━╮┃┗┗┏╯┗┃┃╯┃　　┃┏┣━┓┃┃　┃╯┣━┓       ┃
 * ┃     ╰┫┏━┻╯┗┳┣┛┏┛┃┃┣━━┫┃┃　┃┃┃┗╯　┃　　       ┃
 * ┃     ┏┫━┳━┫┏┃┣┓┗┃┃╯┃　　┃┃┃　┃　┃　┃　┣━┓       ┃
 * ┃     ┗┗┗━━╯┗┛┛╯┗╯╰　┗━━╯　┛　┛┗╯　╰┛┗　　       ┃
 * ┣━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┫
 * ┃                     Copyright (c) 2013 jiangcaiyang                    ┃
 * ┃ This software is provided 'as-is', without any express or implied      ┃
 * ┃ warranty. In no event will the authors be held liable for any damages  ┃
 * ┃ arising from the use of this software.                                 ┃
 * ┃                                                                        ┃
 * ┃ Permission is granted to anyone to use this software for any purpose,  ┃
 * ┃ including commercial applications, and to alter it and redistribute it ┃
 * ┃ freely, subject to the following restrictions:                         ┃
 * ┃                                                                        ┃
 * ┃ 1. The origin of this software must not be misrepresented; you must    ┃
 * ┃    not claim that you wrote the original software. If you use this     ┃
 * ┃    software in a product, an acknowledgment in the product             ┃
 * ┃    documentation would be appreciated but is not required.             ┃
 * ┃ 2. Altered source versions must be plainly marked as such, and must    ┃
 * ┃    not be misrepresented as being the original software.               ┃
 * ┃ 3. This notice may not be removed or altered from any source           ┃
 * ┃    distribution.                                                       ┃
 * ┃                                                                        ┃
 * ┃ jiangcaiyang jiangcaiyang123@163.com                                   ┃
 * ┣━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┫
 * ┃ file name: ShaderOptionDialog.h                                        ┃
 * ┃ create date: 2013年9月24日星期二 18时59分2秒                           ┃
 * ┃ last modified date: 2013年9月24日星期二 22时12分23秒                   ┃
 * ┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛
 */
#ifndef SHADEROPTIONDIALOG_H
#define SHADEROPTIONDIALOG_H

#include <QString>
#include <QMap>
#include <QLineEdit>
#include <QDialog>
#include <QGraphicsProxyWidget>
#include <QState>
#include <QStateMachine>

namespace Ui {
class ShaderOptionDialog;
}

class ShaderOptionDialog : public QDialog
{
    Q_OBJECT
public:
    explicit ShaderOptionDialog(QWidget *parent = 0);
    ~ShaderOptionDialog( void );
signals:
    void SwitchShader( const QString& shaderFileName );
    void SetLightPos( const QVector3D& pos );
private slots:
    void SetSpinValue( int value );
    void SetSliderValue( double value );
    void ShowAxis( bool show );
    void SwitchSelectedShaderText( int row );
    void MakeLightPos( void );
    void ToggleShaderText( bool show );
private:
    void InitShaderList( void );
    void InitEdit( QLineEdit* pEdit );

    Ui::ShaderOptionDialog *ui;

    QMap<QString, QString>      m_ShaderTextMap;
};

#endif // SHADEROPTIONDIALOG_H
